classes/Http/Controller/SignupController.php

```diff
@@ -53,49 +63,43 @@ public function indexAction()
     public function processAction(Request $request)
     {
         try {
-            $this->validate([
+            $this->validate($request, [
                 'email'    => 'required|email',
                 'password' => 'required',
                 'coc'      => 'accepted',
             ]);
 
-            /** @var AccountManagement $accounts */
-            $accounts = $this->service(AccountManagement::class);
-
-            $user = $accounts->create($request->get('email'), $request->get('password'), [
+            $this->accounts->create($request->get('email'), $request->get('password'), [
                 'activated' => 1,
             ]);
-            $accounts->activate($request->get('email'));
+            $this->accounts->activate($request->get('email'));
 
-            $this->app['session']->set('flash', [
+            $request->getSession()->set('flash', [
                 'type'  => 'success',
                 'short' => 'Success',
                 'ext'   => "You've successfully created your account!",
             ]);
 
-            /** @var Authentication $authentication */
-            $authentication = $this->service(Authentication::class);
-
             // Automatically authenticate the newly created user.
-            $authentication->authenticate($request->get('email'), $request->get('password'));
+            $this->authentication->authenticate($request->get('email'), $request->get('password'));
 
             return $this->redirectTo('dashboard');
         } catch (ValidationException $e) {
-            $this->app['session']->set('flash', [
+            $request->getSession()->set('flash', [
                 'type'  => 'error',
                 'short' => $e->getMessage(),
                 'ext'   => $e->errors(),
             ]);
 
-            return $this->redirectBack();
+            return $this->redirectBack($request);
         } catch (\RuntimeException $e) {
-            $this->app['session']->set('flash', [
+            $request->getSession()->set('flash', [
```

Comment:
Since we would otherwise be invoking `$request->getSession()` multiple times, can we maybe just extract a  `$session` variable here?